Client/Assets/SessionRestore.html

```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+ - License, v. 2.0. If a copy of the MPL was not distributed with this
+ - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+ <html>
+ <script>
+    (function () { 
+        var index = document.location.href.search("history");
+        var sessionRestoreComponents = JSON.parse(unescape(document.location.href.substring(index + "history=".length)))
+        var urlList = sessionRestoreComponents['history']
+        var currentPage = sessionRestoreComponents['currentPage']
+        history.replaceState({}, '', '/errors/error.html?url=' + escape(urlList[0]))
+        for (var i = 1; i < urlList.length; i++) {
+            history.pushState({}, '', '/errors/error.html?url=' + escape(urlList[i]))
+        }
+        history.go(currentPage)
```

Comment:
`// We'll end up at the last page pushed, so set the selected index to the current index in the session history.`